lib/compress/zstd_ldm.c

```diff
@@ -286,50 +312,81 @@ static size_t ZSTD_ldm_generateSequences_internal(
     /* Input bounds */
     BYTE const* const istart = (BYTE const*)src;
     BYTE const* const iend = istart + srcSize;
-    BYTE const* const ilimit = iend - MAX(minMatchLength, HASH_READ_SIZE);
+    BYTE const* const ilimit = iend - HASH_READ_SIZE;
     /* Input positions */
     BYTE const* anchor = istart;
     BYTE const* ip = istart;
-    /* Rolling hash */
-    BYTE const* lastHashed = NULL;
-    U64 rollingHash = 0;
-
-    while (ip <= ilimit) {
-        U32 const currentIndex = (U32)(ip - base);
-        U32 hash, checksum;
-        size_t mLength;
-        size_t forwardMatchLength = 0, backwardMatchLength = 0;
-        ldmEntry_t const* bestEntry = NULL;
-        ldmEntry_t newEntry;
-
-        if (ip != istart) {
-            rollingHash = ZSTD_rollingHash_rotate(rollingHash, lastHashed[0],
-                                                  lastHashed[minMatchLength],
-                                                  hashPower);
-        } else {
-            rollingHash = ZSTD_rollingHash_compute(ip, minMatchLength);
-        }
-        lastHashed = ip;
+    /* Rolling hash state */
+    ldmRollingHashState_t hashState;
+    /* Arrays for staged-processing */
+    size_t splits[LDM_LOOKAHEAD_SPLITS];
+    struct {
+        BYTE const* split;
+        U32 hash;
+        U32 checksum;
+        ldmEntry_t* bucket;
+    } candidates[LDM_LOOKAHEAD_SPLITS];
```

Comment:
Same here: Can both of these be moved into the LDM state as well? This is using 2KB of stack space.